src/DynamoCore/Utilities/LuceneSearchUtility.cs

```diff
@@ -25,15 +26,40 @@
 
 namespace Dynamo.Utilities
 {
+    /// <summary>
+    /// Lucene search utility class that will be used for indexing and searching nodes and packages
+    /// </summary>
     internal class LuceneSearchUtility
     {
+        /// <summary>
+        /// Dynamo Model reference
+        /// </summary>
         internal DynamoModel dynamoModel;
```

Comment:
in general I think this can lead to spaghetti code for the following reasons.
1. It's easy to just get whatever references you need from this giant grab bag object without thinking about if it makes sense or theres a better alternative.
2. It's hard for another dev looking at the constructor of this class to figure out what dependencies this class really has.

Reply:
removed